kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/SafeInvoker.java

```diff
@@ -21,10 +24,19 @@
  */
 record SafeInvoker(FilterInvoker invoker) implements FilterInvoker {
 
+    private static final Logger logger = LoggerFactory.getLogger(SafeInvoker.class);
+
     @Override
     public CompletionStage<RequestFilterResult> onRequest(ApiKeys apiKey, short apiVersion, RequestHeaderData header, ApiMessage body, FilterContext filterContext) {
         if (invoker.shouldHandleRequest(apiKey, apiVersion)) {
-            return invoker.onRequest(apiKey, apiVersion, header, body, filterContext);
+            CompletionStage<RequestFilterResult> stage = invoker.onRequest(apiKey, apiVersion, header, body, filterContext);
+            if (stage == null) {
+                if (logger.isDebugEnabled()) {
```

Comment:
I value having this condition logged as a warn.  The user needs to find out about it, so they can fit their code.